sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/windowing/Window.java

```diff
@@ -554,6 +556,46 @@ private Assign(WindowingStrategy updatedStrategy) {
           input.getPipeline(), updatedStrategy, input.isBounded());
     }
 
+    @Override
+    public void populateDisplayData(DisplayData.Builder builder) {
+      super.populateDisplayData(builder);
+
+      builder
```

Comment:
sgtm. Since `original` is only being used for display data, you could make the `Window.Assign` accept a `HasDisplayData` type, and maybe call the parameter/field `displayData` to make its purpose more obvious.